### DISCUSSION TITLE
Different values in Header Component on different pages

### DISCUSSION THREAD

**Discussion by jonathan-franzen** [vercel/next.js#52219] [App Router]
Hi,

I have been looking everywhere, but can't find any good solution. I have a Header component that is active on every page. However, some pages from several different routes etc, needs the padding value in the Header to differ from its initial value. How do I achieve this, and what is the best practice? I have this in my `app/layout.tsx`:

```
export default function RootLayout({children}: {
    children: React.ReactNode;
}) {

    return (
        <html lang="en">
        <body>
        <MenuProvider>
            <CartProvider>
                <StyledComponents>
                    <Header/>
                    <Flow>
                        <main>
                            {children}
                        </main>
                        <Footer/>
                    </Flow>
                    <Menu/>
                    <ShoppingCart/>
                </StyledComponents>
            </CartProvider>
        </MenuProvider>
        </body>
        </html>
    );
}
```
So `app/products/page.tsx` needs to set the header padding to 5px, while `app/products/[slug]/page.tsx` wants the initial value of 60px

**Comment by hesalx** [comment]
1. You can use groups to have different layouts: 
```
app/products/(products)/page.tsx
app/products/(products)/layout.tsx
app/products/(product)/[slug]/page.tsx
app/products/(product)/layout.tsx
```

2. You can use [useSelectedLayoutSegments](https://nextjs.org/docs/app/api-reference/functions/use-selected-layout-segments) inside or as a part of your Header component similar to how active nav links are achieved. 
Assuming `app/layout.tsx`, on `/products` it will be `['products']` and on `/products/slug` it will be `['products', 'slug']`
```jsx
export const Header = () => {
  const segments = useSelectedLayoutSegments()
  const isProductPage = segments.length > 1 && segments[0] === 'products' 
  return <div className={clsx({ [classes.padding]: isProductPage })}>{/* ... */}</div>
}
```

3. You can use a CSS trick that lets you style anything on the page based on what's anywhere on the page depending on your compatibility requirements: https://caniuse.com/css-has 

In your header:
```css
:root:has(.headerPadded) .header { /* ... */ }
```
In your `app/products/[slug]/page.tsx`
```html
<span className="headerPadded" />
```
Of course, you can always have the above snippet as a part of you Header component/module such that it uses a generated class name and can be re-used:
```jsx
export const Padded = () => <span className={classes.padded} />
```
```jsx
export default function Page () {
   return <>
     <Header.Padded />
     {/* ... */}
   </>
}
